src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const call = (...args) => {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += args[0];
```

Comment:
It's good practice to explicitly check for `undefined` when you expect a function to be called without arguments. This makes the intent clearer and the code more robust.

Reply:
[comment](https://github.com/mate-academy/js_task-inifinite-sum-closures/pull/2028/files#r1683618132) not fixed